crates/solver/src/interactions/zeroex.rs

```diff
@@ -0,0 +1,41 @@
+use crate::{encoding::EncodedInteraction, settlement::Interaction};
+use contracts::IZeroEx;
+use ethcontract::Bytes;
+use shared::zeroex_api::Order;
+
+#[derive(Clone, Debug)]
+pub struct ZeroExInteraction {
+    pub order: Order,
+    pub executed_amount: u128,
+    pub zeroex: IZeroEx,
+}
+
+impl Interaction for ZeroExInteraction {
+    fn encode(&self) -> Vec<EncodedInteraction> {
+        let method = self.zeroex.fill_or_kill_limit_order(
```

Comment:
Why do you use `fill_or_kill` here if the orders are partially fillable?

**edit**: The 0x function name threw me off: https://github.com/0xProject/protocol/blob/development/docs/basics/functions.rst#fillorkilllimitorder

the "fill-or-kill` means that it fills the entire _specified executed amount_ and **not** the _entire remaining amount_ of order. The alternate `fillLimitOrder` method allows the order to be partially filled **up to** the specified amount.

Reply:
I must have read the docs not carefully enough. 🤦‍♂️ 
Will change it.

Reply:
Nono! It is correct as is!

Reply:
Arrgh. Just pushed the change. 😅 
Will revert it again.